networkx/utils/misc.py

```diff
@@ -175,3 +175,11 @@ def dict_to_numpy_array1(d,mapping=None):
         i = mapping[k1]
         a[i] = d[k1]
     return a
+
+def iterlen(iterator):
+    """ Returns the length of the iterator.
+
+    This will exhaust the iterator, use carefully.
```

Comment:
Giving example code here for why this is "dangerous" would be good, something like

    >>> i = iter([1, 2, 3])
    >>> iterlen(i)
    3
    >>> next(i)
    # show the error message here